src/mlpack/tests/feedforward_network_test.cpp

```diff
@@ -57,6 +57,93 @@ void TestNetwork(ModelType& model,
   REQUIRE(classificationError <= classificationErrorThreshold);
 }
 
+// network1 should be allocated with `new`, and trained on some data.
+template<typename MatType = arma::mat, typename ModelType>
+void CheckCopyFunction(ModelType* network1,
+                       MatType& trainData,
+                       MatType& trainLabels,
+                       MatType& testData,
+                       MatType& testLabels,
+                       const size_t maxEpochs,
+                       const double classificationErrorThreshold)
```

Comment:
Do you think we could make another version of this function that uses the move constructor instead?  e.g. `FFN<> network2(std::move(*network1));`?  Then this could also test the move constructor and operator functionality.

Reply:
I added this in same function.

Reply:
Actually I thought about it and I think maybe it would be better to have this in a separate function.  If we call the move constructor, `network1` should be "empty", so when we call `delete network1;` it's possible that `network3` will still be pointing to some of `network1`'s original memory and so that memory won't be deleted.  i.e., the goal here is to check that `network2` doesn't use any of `network1`'s memory by deallocating `network1` after copy.  But if we use the move constructor, then we actually don't end up deallocating `network1`'s memory.  I hope this makes sense... let me know if I can clarify.

Reply:
Yeah got it added separate function.